ui/view/board.py

```diff
@@ -37,6 +42,55 @@ def point_to_hex_with_board_offsets(pos):
     return Hex(cell[0] + BOARD_MAX_COLS // 2, cell[1])
 
 
+class MarbleAnim(ABC):
+    """
+    A generic marble animation.
+    Defines the `transform` method for composing marble drawing parameters.
+    For more complex operands, consider wrapping drawing parameters into their
+    own class.
+    """
+
+    @abstractmethod
+    def transform(self, cell, size):
+        """
+        Transforms the given marble drawing parameters using the animation state.
+        :param cell: a Hex
+        :param size: a float
+        :return: a tuple[Hex, float]
+        """
+
+class MarbleMoveAnim(MarbleAnim, HexTweenAnim):
+    """
+    A marble movement animation.
+    Usage requires a `src` and `dest` per `HexTweenAnim`.
+    """
+    duration = 10
+
+    def transform(self, _, size):
+        """
+        Transforms the given marble drawing parameters using the animation state.
+        :param cell: a Hex
+        :param size: a float
+        :return: a tuple[Hex, float]
+        """
+        return self.cell, size
+
+class MarbleShrinkAnim(MarbleAnim, TweenAnim):
+    """
+    A marble shrink animation.
+    """
+    duration = 7
+
+    def transform(self, cell, size):
+        """
+        Transforms the given marble drawing parameters using the animation state.
+        :param cell: a Hex
+        :param size: a float
+        :return: a tuple[Hex, float]
+        """
+        return cell, size * (1 - self.pos)
+
+
```

Comment:
This is a nit but I wasn't sure whether to keep this in the marble view or not. I guess it only pertains to the marbles themselves but the animations don't really have any utility outside of the board view. It'd probably make more sense to move it in if the board was drawing stuff other than the marbles